<launch>
  <node pkg="nodelet" type="nodelet" name="cmd_vel_mux" args="load cmd_vel_mux/CmdVelMuxNodelet mobile_base_nodelet_manager">
    <param name="yaml_cfg_file" value="$(find kobuki_keyop)/param/keyop_mux.yaml"/>
    <remap from="cmd_vel_mux/output" to="/keyop_vel_smoother/raw_cmd_vel"/>
  </node>
  
  <node pkg="nodelet" type="nodelet" name="kobuki_safety_controller" args="load kobuki_safety_controller/SafetyControllerNodelet mobile_base_nodelet_manager">
    <remap from="kobuki_safety_controller/cmd_vel" to="cmd_vel_mux/safety_controller"/>
    <remap from="kobuki_safety_controller/events/bumper" to="mobile_base/events/bumper"/>
    <remap from="kobuki_safety_controller/events/cliff" to="mobile_base/events/cliff"/>
    <remap from="kobuki_safety_controller/events/wheel_drop" to="mobile_base/events/wheel_drop"/>
  </node>
  
  <node pkg="nodelet" type="nodelet" name="keyop_vel_smoother" args="load yocs_velocity_smoother/VelocitySmootherNodelet mobile_base_nodelet_manager">
    <rosparam file="$(find rp_locomotion)/param/keyop_smoother.yaml" command="load"/>
    <remap from="keyop_vel_smoother/odometry" to="odom"/>
    <remap from="keyop_vel_smoother/smooth_cmd_vel" to="mobile_base/commands/velocity"/>
  </node>
  
  <node ns="rp" name="navigation_node" pkg="rp_navigation" type="rp_navigation_node"/>
  
  <node ns="rp" name="locomotion_node" pkg="rp_locomotion" type="rp_locomotion_node" output="screen">
    <remap from="locomotion/velocity" to="/cmd_vel_mux/keyboard_teleop"/>
    <remap from="locomotion/motor_power" to="/mobile_base/commands/motor_power"/>
    <param name="framing_linear_velocity" value="0.1" type="double"/>
    <param name="framing_angular_velocity" value="0.33" type="double"/>
    <param name="obstacle_avoidance_linear_velocity" value="0.15" type="double"/>
    <param name="obstacle_avoidance_angular_velocity" value="0.5" type="double"/>    
  </node>
</launch>
